src/HandHistory.js

```diff
@@ -0,0 +1,148 @@
+class HandHistory {
+
+    constructor(player, cards) {
+        this.actions = [];
+        this.hero = player;
+        this.heroCards = cards;
+    }
+
+    // After you build one then for fuck's sake don't modify it. I'm trusting you.
+
+    // Actions
+
+    addStack(player, amount) {
+        this.actions.push({action: 'stack', player: player, amount: amount});
+    }
+
+    addBet(player, amount) {
+        this.actions.push({action: 'bet', player: player, amount: amount});
+    }
+
+    fold(player) {
+        this.actions.push({action: 'fold', player: player});
+    }
+
+    check(player) {
+        this.actions.push({action: 'check', player: player});
+    }
+
+    dealFlop(cards) {
+        this.actions.push({action: 'flop', cards: cards});
+    }
+
+    dealTurn(card) {
+        this.actions.push({action: 'turn', card: card});
+    }
+
+    dealRiver(card) {
+        this.actions.push({action: 'river', card: card});
+    }
+
+    actionOn(player) {
+        this.actionOn = player;
+    }
+
+    // Queries
+
+    stack(player) {
+        var result = 0;
+        this.actionsFor(player).forEach(each => {
+            switch(each.action) {
+                case 'stack':
+                    result += each.amount;
+                    break;
+                case 'bet':
+                    result -= each.amount;
+                    break;
+            }
+        });
+        return result;
+    }
+
+    pot() {
+        const bets = this.actions.filter(each => each.action === 'bet');
+        return bets.reduce((prev, each) => prev + each.amount, 0);
+    }
+
+    players() {
+        const stacks = this.actions.filter(each => each.action === 'stack');
+        return stacks.map(each => each.player);
+    }
+    
+    cards() {
+        return this.heroCards;
+    }
+    
+    isHero(player) {
```

Comment:
There seem to be two kinds of methods here: things that ask a question about a specific player, versus things that present global information.